src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetVersionsControllerTests.cs

```diff
@@ -0,0 +1,299 @@
+                    .Generate(1))
+                .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.Add(dataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            var processorClient = new Mock<IProcessorClient>();
+            processorClient
+                .Setup(c => c.DeleteDataSetVersion(
+                    dataSetVersion.Id,
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new Either<ActionResult, Unit>(new NotFoundResult()));
+
+            var client = BuildApp(processorClient.Object).CreateClient();
+
+            var response = await DeleteVersion(dataSetVersion.Id, client);
+
+            response.AssertNotFound();
+        }
+
+        [Fact]
+        public async Task ProcessorReturns400_Returns400()
+        {
+            DataSet dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusDraft();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion(filters: 1, indicators: 1, locations: 1, timePeriods: 2)
+                .WithVersionNumber(1, 0, 0)
+                .WithStatusDraft()
+                .WithDataSet(dataSet)
+                .WithImports(() => DataFixture
+                    .DefaultDataSetVersionImport()
+                    .Generate(1))
+                .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.Add(dataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            var processorClient = new Mock<IProcessorClient>();
+            processorClient
+                .Setup(c => c.DeleteDataSetVersion(
+                    dataSetVersion.Id,
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new Either<ActionResult, Unit>(
+                    new BadRequestObjectResult(
+                        new ValidationProblemViewModel
+                        {
+                            Errors = new ErrorViewModel[]
+                                {
+                                    new() {
+                                       Code = Errors.Error1.ToString()
+                                    }
+                                }
+                        })));
+
+            var client = BuildApp(processorClient.Object).CreateClient();
+
+            var response = await DeleteVersion(dataSetVersion.Id, client);
+
+            var validationProblem = response.AssertValidationProblem();
+
+            var error = validationProblem.AssertHasError(null, Errors.Error1.ToString());
```

Comment:
As these are both string params I think it would be a good idea to label them, and it would give an indication to the reader to say what this `null` represents (i.e. the path) without having to inspect it.

Reply:
I've just gotten rid of the `Enum` and hardcoded some strings for the `Code` and `Path` :)